--- a/decoupled/60/0.8_0.2/25/django_django/tests/foreign_object/test_empty_join.py
+++ b/decoupled/60/0.8_0.2/25/django_django/tests/foreign_object/test_empty_join.py
@@ -41,7 +41,7 @@
             ['a', 'a/b', 'a/b/a'],

         )

 

-    def test_empty_join_conditions(self):

+def test_empty_join_conditions(self):

         x = SlugPage.objects.get(slug='x')

         message = "Join generated an empty ON clause."

         with self.assertRaisesMessage(ValueError, message):
