--- a/decoupled/60/0.8_0.2/164/pylint-dev_pylint/tests/functional/a/access/access_attr_before_def_false_positive.py
+++ b/decoupled/60/0.8_0.2/164/pylint-dev_pylint/tests/functional/a/access/access_attr_before_def_false_positive.py
@@ -26,16 +26,16 @@
         """

         telnetlib.Telnet.__init__(self, host, port)

 

-    def readUntilArray(self, matches, _=None):

-        """

-        Read until a given string is encountered or until timeout.

-        ...

-        """

-        self.process_rawq()

-        maxLength = 0

-        for match in matches:

-            if len(match) > maxLength:

-                maxLength = len(match)

+def readUntilArray(self, matches, _=None):

+    """

+    Read until a given string is encountered or until timeout.

+    ...

+    """

+    self.process_rawq()

+    maxLength = 0

+    for match in matches:

+        if len(match) > maxLength:

+            maxLength = len(match)

 

 class Base(object):

     """bla bla"""
