--- a/decoupled/60/0.8_0.2/115/django_django/tests/deprecation/tests.py
+++ b/decoupled/60/0.8_0.2/115/django_django/tests/deprecation/tests.py
@@ -31,12 +31,15 @@
                 def old(self):

                     pass

 

-    def test_get_new_defined(self):

+def test_get_new_defined(self):

         """

         Ensure `old` complains and not `new` when only `new` is defined.

         """

 

         class Manager(metaclass=RenameManagerMethods):

+            def old(self):

+                warnings.warn("`Manager.old` is deprecated, use `new` instead.", DeprecationWarning)

+

             def new(self):

                 pass

 
