--- a/decoupled/60/0.8_0.2/138/matplotlib_matplotlib/galleries/examples/widgets/check_buttons.py
+++ b/decoupled/60/0.8_0.2/138/matplotlib_matplotlib/galleries/examples/widgets/check_buttons.py
@@ -41,10 +41,12 @@
 )

 

 

-def callback(label):

-    ln = lines_by_label[label]

-    ln.set_visible(not ln.get_visible())

-    ln.figure.canvas.draw_idle()

+def onchanged(values):

+    print("on changed")

+    print(values)

+    pyplot.clf()

+    addElements()

+    pyplot.canvas.draw_idle()

 

 check.on_clicked(callback)

 
