--- a/decoupled/60/0.8_0.2/242/sympy_sympy/sympy/core/numbers.py
+++ b/decoupled/60/0.8_0.2/242/sympy_sympy/sympy/core/numbers.py
@@ -79,7 +79,7 @@
 def mpf_norm(mpf, prec):

     """Return the mpf tuple normalized appropriately for the indicated

     precision after doing a check to see if zero should be returned or

-    not when the mantissa is 0. ``mpf_normlize`` always assumes that this

+    not when the mantissa is 0. ``mpf_normalize`` always assumes that this

     is zero, but it may not be since the mantissa for mpf's values "+inf",

     "-inf" and "nan" have a mantissa of zero, too.

 
