--- a/decoupled/60/0.8_0.2/201/scikit-learn_scikit-learn/sklearn/cluster/_spectral.py
+++ b/decoupled/60/0.8_0.2/201/scikit-learn_scikit-learn/sklearn/cluster/_spectral.py
@@ -18,7 +18,7 @@
 from ._k_means import k_means

 

 

-def discretize(vectors, copy=True, max_svd_restarts=30, n_iter_max=20,

+def discretize(vectors, copy=True, max_svd_restarts=30, n_iter_max=30,

                random_state=None):

     """Search for a partition matrix (clustering) which is closest to the

     eigenvector embedding.

@@ -115,45 +115,7 @@
         # possible

         c = np.zeros(n_samples)

         for j in range(1, n_components):

-            # Accumulate c to ensure row is as orthogonal as possible to

-            # previous picks as well as current one

-            c += np.abs(np.dot(vectors, rotation[:, j - 1]))

-            rotation[:, j] = vectors[c.argmin(), :].T

-

-        last_objective_value = 0.0

-        n_iter = 0

-

-        while not has_converged:

-            n_iter += 1

-

-            t_discrete = np.dot(vectors, rotation)

-

-            labels = t_discrete.argmax(axis=1)

-            vectors_discrete = csc_matrix(

-                (np.ones(len(labels)), (np.arange(0, n_samples), labels)),

-                shape=(n_samples, n_components))

-

-            t_svd = vectors_discrete.T * vectors

-

-            try:

-                U, S, Vh = np.linalg.svd(t_svd)

-                svd_restarts += 1

-            except LinAlgError:

-                print("SVD did not converge, randomizing and trying again")

-                break

-

-            ncut_value = 2.0 * (n_samples - S.sum())

-            if ((abs(ncut_value - last_objective_value) < eps) or

-                    (n_iter > n_iter_max)):

-                has_converged = True

-            else:

-                # otherwise calculate rotation and continue

-                last_objective_value = ncut_value

-                rotation = np.dot(Vh.T, U.T)

-

-    if not has_converged:

-        raise LinAlgError('SVD did not converge')

-    return labels

+            # Accumulate c to

 

 

 def spectral_clustering(affinity, n_clusters=8, n_components=None,
