--- a/decoupled/60/0.8_0.2/41/django_django/tests/delete/tests.py
+++ b/decoupled/60/0.8_0.2/41/django_django/tests/delete/tests.py
@@ -673,7 +673,7 @@
         self.assertFalse(Parent.objects.exists())

         self.assertFalse(Child.objects.exists())

 

-    def test_fast_delete_large_batch(self):

+def test_fast_delete_large_batch(self):

         User.objects.bulk_create(User() for i in range(0, 2000))

         # No problems here - we aren't going to cascade, so we will fast

         # delete the objects in a single query.
