--- a/decoupled/60/0.8_0.2/216/sphinx-doc_sphinx/sphinx/util/typing.py
+++ b/decoupled/60/0.8_0.2/216/sphinx-doc_sphinx/sphinx/util/typing.py
@@ -58,10 +58,10 @@
 

 

 def get_type_hints(obj: Any, globalns: Dict = None, localns: Dict = None) -> Dict[str, Any]:

-    """Return a dictionary containing type hints for a function, method, module or class object.

-

-    This is a simple wrapper of `typing.get_type_hints()` that does not raise an error on

-    runtime.

+    """Return a dictionary containing type hints for a function, method, module, or class object.

+

+    This is a simple wrapper of `typing.get_type_hints()` that handles various exceptions that may occur

+    during the evaluation of type hints, such as `NameError`, `TypeError`, `KeyError`, and `AttributeError`.

     """

     from sphinx.util.inspect import safe_getattr  # lazy loading

 
