--- a/decoupled/60/0.8_0.2/110/django_django/django/db/models/query_utils.py
+++ b/decoupled/60/0.8_0.2/110/django_django/django/db/models/query_utils.py
@@ -110,7 +110,7 @@
             else:

                 yield child

 

-    def check(self, against, using=DEFAULT_DB_ALIAS):

+def check(self, against, using=DEFAULT_DB_ALIAS):

         """

         Do a database query to check if the expressions of the Q instance

         matches against the expressions.

@@ -134,7 +134,7 @@
             query.add_q(self)

         compiler = query.get_compiler(using=using)

         try:

-            return compiler.execute_sql(SINGLE) is not None

+            return compiler.execute_sql(SINGLE, using=using) is not None

         except DatabaseError as e:

             logger.warning("Got a database error calling check() on %r: %s", self, e)

             return True
