--- a/decoupled/60/0.8_0.2/207/sphinx-doc_sphinx/sphinx/ext/autodoc/type_comment.py
+++ b/decoupled/60/0.8_0.2/207/sphinx-doc_sphinx/sphinx/ext/autodoc/type_comment.py
@@ -16,15 +16,10 @@
 

 def not_suppressed(argtypes: List[ast.AST] = []) -> bool:

     """Check given *argtypes* is suppressed type_comment or not."""

-    if len(argtypes) == 0:  # no argtypees

+    if len(argtypes) == 1 and isinstance(argtypes[0], ast.Ellipsis):  # suppressed

+        return True  # Corrected to return True for a single Ellipsis

+    else:  # not suppressed

         return False

-    elif len(argtypes) == 1 and ast_unparse(argtypes[0]) == "...":  # suppressed

-        # Note: To support multiple versions of python, this uses ``ast_unparse()`` for

-        # comparison with Ellipsis.  Since 3.8, ast.Constant has been used to represent

-        # Ellipsis node instead of ast.Ellipsis.

-        return False

-    else:  # not suppressed

-        return True

 

 

 def signature_from_ast(node: ast.FunctionDef, bound_method: bool,
