--- a/decoupled/60/0.8_0.2/36/django_django/django/db/models/fields/related_descriptors.py
+++ b/decoupled/60/0.8_0.2/36/django_django/django/db/models/fields/related_descriptors.py
@@ -380,7 +380,7 @@
             self.related.field.set_cached_value(rel_obj, instance)

         return queryset, rel_obj_attr, instance_attr, True, self.related.get_cache_name(), False

 

-    def __get__(self, instance, cls=None):

+def __get__(self, instance, cls=None):

         """

         Get the related instance through the reverse relation.

 
