--- a/decoupled/60/0.8_0.2/33/django_django/django/db/backends/base/introspection.py
+++ b/decoupled/60/0.8_0.2/33/django_django/django/db/backends/base/introspection.py
@@ -14,15 +14,13 @@
     def __init__(self, connection):

         self.connection = connection

 

-    def get_field_type(self, data_type, description):

-        """

-        Hook for a database backend to use the cursor description to

-        match a Django field type to a database column.

-

-        For Oracle, the column data_type on its own is insufficient to

-        distinguish between a FloatField and IntegerField, for example.

-        """

-        return self.data_types_reverse[data_type]

+# settings.py

+LANGUAGES = [

+    ('de-at', _('Austrian German')),

+    ('de', _('German')),

+    # other languages...

+]

+LANGUAGE_CODE = 'de-at'

 

     def identifier_converter(self, name):

         """
