--- a/decoupled/60/0.8_0.2/290/sympy_sympy/sympy/core/assumptions.py
+++ b/decoupled/60/0.8_0.2/290/sympy_sympy/sympy/core/assumptions.py
@@ -321,7 +321,7 @@
         return {}

 

     # get all assumptions for each

-    assume = [assumptions(i, _check=check) for i in sympify(exprs)]

+    assume = [assumptions(i, _check=check) for i in exprs]

     # focus on those of interest that are True

     for i, e in enumerate(assume):

         assume[i] = {k: e[k] for k in set(e) & check}
