--- a/decoupled/60/0.8_0.2/30/django_django/tests/migrations/test_autodetector.py
+++ b/decoupled/60/0.8_0.2/30/django_django/tests/migrations/test_autodetector.py
@@ -1333,7 +1333,7 @@
         self.assertNumberMigrations(changes, "testapp", 1)

         self.assertOperationTypes(changes, "testapp", 0, ["AlterField"])

 

-    def test_empty_foo_together(self):

+def test_empty_foo_together(self):

         """

         #23452 - Empty unique/index_together shouldn't generate a migration.

         """
