--- a/decoupled/60/0.8_0.2/137/matplotlib_matplotlib/lib/matplotlib/backend_tools.py
+++ b/decoupled/60/0.8_0.2/137/matplotlib_matplotlib/lib/matplotlib/backend_tools.py
@@ -484,17 +484,24 @@
         self.home_views = WeakKeyDictionary()

         super().__init__(*args, **kwargs)

 

-    def add_figure(self, figure):

+def add_figure(self, figure):

         """Add the current figure to the stack of views and positions."""

+        # Save the current WeakKeyDictionary

+        home_views_backup = figure.home_views

+

+        # Remove the WeakKeyDictionary before pickling

+        figure.home_views = {}

 

         if figure not in self.views:

             self.views[figure] = cbook.Stack()

             self.positions[figure] = cbook.Stack()

-            self.home_views[figure] = WeakKeyDictionary()

             # Define Home

             self.push_current(figure)

             # Make sure we add a home view for new axes as they're added

             figure.add_axobserver(lambda fig: self.update_home_views(fig))

+

+        # Restore the WeakKeyDictionary after pickling

+        figure.home_views = home_views_backup

 

     def clear(self, figure):

         """Reset the axes stack."""
