--- a/decoupled/60/0.8_0.2/6/django_django/django/core/files/move.py
+++ b/decoupled/60/0.8_0.2/6/django_django/django/core/files/move.py
@@ -27,6 +27,11 @@
             os.path.normcase(os.path.abspath(dst)))

 

 

+import os

+import stat

+import errno

+from django.conf import settings

+

 def file_move_safe(old_file_name, new_file_name, chunk_size=1024 * 64, allow_overwrite=False):

     """

     Move a file from one location to another in the safest way possible.

@@ -46,6 +51,8 @@
             raise FileExistsError('Destination file %s exists and allow_overwrite is False.' % new_file_name)

 

         os.rename(old_file_name, new_file_name)

+        # Set the default file permissions to 0o644

+        os.chmod(new_file_name, 0o644)

         return

     except OSError:

         # OSError happens with os.rename() if moving to another filesystem or

@@ -85,3 +92,6 @@
         # on close anyway.)

         if getattr(e, 'winerror', 0) != 32:

             raise

+

+    # Set the default file permissions to 0o644

+    os.chmod(new_file_name, 0o644)
