--- a/decoupled/60/0.8_0.2/23/django_django/django/db/models/query_utils.py
+++ b/decoupled/60/0.8_0.2/23/django_django/django/db/models/query_utils.py
@@ -119,7 +119,7 @@
     def __init__(self, field):

         self.field = field

 

-    def __get__(self, instance, cls=None):

+def __get__(self, instance, cls=None):

         """

         Retrieve and caches the value from the datastore on the first lookup.

         Return the cached value.

@@ -135,7 +135,8 @@
             if val is None:

                 instance.refresh_from_db(fields=[field_name])

                 val = getattr(instance, field_name)

-            data[field_name] = val

+            # Ensure we cache the name of the Enum object, not its value

+            data[field_name] = val.name if isinstance(val, Enum) else val

         return data[field_name]

 

     def _check_parent_chain(self, instance):
