--- a/decoupled/60/0.8_0.2/190/scikit-learn_scikit-learn/sklearn/utils/estimator_checks.py
+++ b/decoupled/60/0.8_0.2/190/scikit-learn_scikit-learn/sklearn/utils/estimator_checks.py
@@ -2000,6 +2000,15 @@
 

 

 @ignore_warnings(category=(DeprecationWarning, FutureWarning))

+import numpy as np

+from sklearn.mixture import GaussianMixture

+from sklearn.utils import check_array

+from sklearn.externals import six

+from sklearn.utils.validation import check_is_fitted

+from sklearn.utils.sparsefuncs import csc_median_axis_0

+from scipy.sparse import csc_matrix

+from sklearn.utils import pickle

+

 def check_sparsify_coefficients(name, estimator_orig):

     X = np.array([[-2, -1], [-1, -1], [-1, -2], [1, 1], [1, 2], [2, 1],

                   [-1, -2], [2, 2], [-2, -2]])
