--- a/decoupled/60/0.8_0.2/77/django_django/django/db/models/fields/related_descriptors.py
+++ b/decoupled/60/0.8_0.2/77/django_django/django/db/models/fields/related_descriptors.py
@@ -287,7 +287,7 @@
     ``Restaurant.place`` is a ``ForwardOneToOneDescriptor`` instance.

     """

 

-    def get_object(self, instance):

+def get_object(self, instance):

         if self.field.remote_field.parent_link:

             deferred = instance.get_deferred_fields()

             # Because it's a parent link, all the data is available in the

@@ -304,6 +304,8 @@
                 obj._state.adding = instance._state.adding

                 obj._state.db = instance._state.db

                 return obj

+        # Add missing make_hashable call on self.through_fields

+        self.through_fields = make_hashable(self.through_fields)

         return super().get_object(instance)

 

     def __set__(self, instance, value):
