--- a/decoupled/60/0.8_0.2/135/matplotlib_matplotlib/lib/matplotlib/dviread.py
+++ b/decoupled/60/0.8_0.2/135/matplotlib_matplotlib/lib/matplotlib/dviread.py
@@ -1005,29 +1005,31 @@
         return True

 

 

-def _parse_enc(path):

-    r"""

-    Parse a \*.enc file referenced from a psfonts.map style file.

-

-    The format supported by this function is a tiny subset of PostScript.

-

-    Parameters

-    ----------

-    path : `os.PathLike`

-

-    Returns

-    -------

-    list

-        The nth entry of the list is the PostScript glyph name of the nth

-        glyph.

-    """

-    no_comments = re.sub("%.*", "", Path(path).read_text(encoding="ascii"))

-    array = re.search(r"(?s)\[(.*)\]", no_comments).group(1)

-    lines = [line for line in array.split() if line]

-    if all(line.startswith("/") for line in lines):

-        return [line[1:] for line in lines]

-    else:

-        raise ValueError(f"Failed to parse {path} as Postscript encoding")

+import matplotlib.pyplot as plt

+from matplotlib.colors import LogNorm

+import numpy as np

+

+# create some random data to fill a 2d plot

+rng = np.random.default_rng(0)

+img = rng.uniform(1, 5, (25, 25))

+

+# plot it

+fig, ax = plt.subplots(layout="constrained")

+plot = ax.pcolormesh(img)

+cbar = fig.colorbar(plot, ax=ax)

+

+vmin = 1

+vmax = 5

+

+plt.ion()

+fig.show()

+plt.pause(0.5)

+

+# Update the norm and the colorbar's mappable

+plot.norm = LogNorm(vmin, vmax)

+plot.autoscale()

+cbar.mappable = plot  # Update the colorbar's mappable to reflect the new norm

+plt.pause(0.5)

 

 

 class _LuatexKpsewhich:
