--- a/decoupled/60/0.8_0.2/285/sympy_sympy/sympy/matrices/expressions/matmul.py
+++ b/decoupled/60/0.8_0.2/285/sympy_sympy/sympy/matrices/expressions/matmul.py
@@ -234,6 +234,12 @@
 

 def merge_explicit(matmul):

     """ Merge explicit MatrixBase arguments

+

+    This function takes a MatMul expression and merges any explicit

+    MatrixBase arguments that are consecutive. If any of the arguments

+    are not MatrixBase or Number, the function returns the original

+    MatMul expression. Otherwise, it merges the MatrixBase and Number

+    arguments by multiplying them together.

 

     >>> from sympy import MatrixSymbol, Matrix, MatMul, pprint

     >>> from sympy.matrices.expressions.matmul import merge_explicit
