--- a/decoupled/60/0.8_0.2/274/sympy_sympy/sympy/matrices/solvers.py
+++ b/decoupled/60/0.8_0.2/274/sympy_sympy/sympy/matrices/solvers.py
@@ -605,8 +605,7 @@
         for that purpose; if so, it must be the same shape as x, with as

         many rows as matrix A has columns, and as many columns as matrix

         B.  If left as None, an appropriate matrix containing dummy

-        symbols in the form of ``wn_m`` will be used, with n and m being

-        row and column position of each symbol.

+        symbols in the form of ``w0_0``, ``w0_1``, etc. will be used.

 

     Returns

     =======

@@ -651,29 +650,6 @@
     This may return either exact solutions or least squares solutions.

     To determine which, check ``A * A.pinv() * B == B``.  It will be

     True if exact solutions exist, and False if only a least-squares

-    solution exists.  Be aware that the left hand side of that equation

-    may need to be simplified to correctly compare to the right hand

-    side.

-

-    References

-    ==========

-

-    .. [1] https://en.wikipedia.org/wiki/Moore-Penrose_pseudoinverse#Obtaining_all_solutions_of_a_linear_system

-

-    """

-

-    from sympy.matrices import eye

-

-    A      = M

-    A_pinv = M.pinv()

-

-    if arbitrary_matrix is None:

-        rows, cols       = A.cols, B.cols

-        w                = symbols('w:{}_:{}'.format(rows, cols), cls=Dummy)

-        arbitrary_matrix = M.__class__(cols, rows, w).T

-

-    return A_pinv.multiply(B) + (eye(A.cols) -

-            A_pinv.multiply(A)).multiply(arbitrary_matrix)

 

 

 def _solve(M, rhs, method='GJ'):
