--- a/decoupled/60/0.8_0.2/28/django_django/django/db/models/sql/query.py
+++ b/decoupled/60/0.8_0.2/28/django_django/django/db/models/sql/query.py
@@ -251,16 +251,16 @@
         for alias in self.alias_map:

             return alias

 

-    def __str__(self):

-        """

-        Return the query as a string of SQL with the parameter values

-        substituted in (use sql_with_params() to see the unsubstituted string).

-

-        Parameter values won't necessarily be quoted correctly, since that is

-        done by the database interface at execution time.

-        """

-        sql, params = self.sql_with_params()

-        return sql % params

+def __str__(self):

+    """

+    Return the query as a string of SQL with the parameter values

+    substituted in (use sql_with_params() to see the unsubstituted string).

+

+    Parameter values won't necessarily be quoted correctly, since that is

+    done by the database interface at execution time.

+    """

+    sql, params = self.sql_with_params()

+    return sql % params

 

     def sql_with_params(self):

         """
