--- a/decoupled/60/0.8_0.2/12/django_django/tests/auth_tests/test_checks.py
+++ b/decoupled/60/0.8_0.2/12/django_django/tests/auth_tests/test_checks.py
@@ -55,7 +55,7 @@
         ])

 

     @override_settings(AUTH_USER_MODEL='auth_tests.CustomUserNonUniqueUsername')

-    def test_username_non_unique(self):

+def test_username_non_unique(self):

         """

         A non-unique USERNAME_FIELD raises an error only if the default

         authentication backend is used. Otherwise, a warning is raised.
