--- a/decoupled/60/0.8_0.2/76/django_django/django/db/models/fields/related_descriptors.py
+++ b/decoupled/60/0.8_0.2/76/django_django/django/db/models/fields/related_descriptors.py
@@ -153,7 +153,7 @@
         # Assuming the database enforces foreign keys, this won't fail.

         return qs.get(self.field.get_reverse_related_filter(instance))

 

-    def __get__(self, instance, cls=None):

+def __get__(self, instance, cls=None):

         """

         Get the related instance through the forward relation.

 
