--- a/decoupled/60/0.8_0.2/128/matplotlib_matplotlib/lib/matplotlib/tests/test_legend.py
+++ b/decoupled/60/0.8_0.2/128/matplotlib_matplotlib/lib/matplotlib/tests/test_legend.py
@@ -921,12 +921,14 @@
     assert handles[0].get_marker() == leg.legendHandles[0].get_marker()

 

 

+import matplotlib.pyplot as plt

+

 def test_subfigure_legend():

-    # Test that legend can be added to subfigure (#20723)

+    # Test that a draggable legend can be added to subfigure

     subfig = plt.figure().subfigures()

     ax = subfig.subplots()

     ax.plot([0, 1], [0, 1], label="line")

-    leg = subfig.legend()

+    leg = subfig.legend(draggable=True)

     assert leg.figure is subfig

 

 
